### ISSUE TITLE
ERROR: Failed to parse JSON (issue/danger/github) 

### ISSUE THREAD

**Issue by TheiOSDude** [danger/swift#316]
Hey

Trying to run danger-swift on Github's actions, but the following error:

```
ERROR: Failed to parse JSON: keyNotFound(CodingKeys(stringValue: "issue", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "danger", intValue: nil), CodingKeys(stringValue: "github", intValue: nil)], debugDescription: "No value associated with key CodingKeys(stringValue: \"issue\", intValue: nil) (\"issue\").", underlyingError: nil))

ERROR: Dangerfile eval failed at Dangerfile.swift

Danger: ⅹ Failing the build, there is 1 fail.
## Failures
danger-swift` failed.
```

Any help would be greatly appreciated.
Danger file is
```
import Danger

SwiftLint.lint()
```

**Comment by orta** [comment]
also, are you on a GitHub enterprise account? I've never seen an issue being optional in a PR (in my non-enterprise world)

**Comment by f-meloni** [comment]
It should use `pull_request` as trigger, that was my though, I think the `event.json` doesn't contain the pull request data otherwise

**Comment by orta** [comment]
Yeah, that's true - you need `danger-swift local` to run when it's not a PR

**Comment by orta** [comment]
@f-meloni probably worth a unique error message in this case?

**Comment by orta** [comment]
That could actually live in danger-js too

**Comment by f-meloni** [comment]
@orta I remember I've added this
https://github.com/danger/danger-js/blob/master/source/ci_source/providers/GitHubActions.ts#L165-L187
To check if the event.json was valid, but apparently the event.json (also looking at the crash message) of that push event, has a pullRequest object with values on it, thing that is really strange, and not mentioned on PushEvent doc https://developer.github.com/v3/activity/events/types/#pushevent
We could maybe use `GITHUB_EVENT_NAME` and check if the event name is one of the supported ones, that I think are:
- `pull_request`
- `pull_request_review`
- `pull_request_review_comment`
- `issue_comment_event`

**Comment by orta** [comment]
hah, nice work, I think that's probably a good call

**Comment by TheiOSDude** [comment]
Thanks, you early suggestion got it moving and not having this issue. The next issue for me to resolve it `/bin/sh: 1: swiftlint: not found` :/

**Comment by f-meloni** [comment]
Yes, that is because swiftlint is really heavy to build, then github actions doesn't build it for you.
There are several alternatives:
- Release a swiftlint dockerfile as part of the release process as mentioned on https://github.com/danger/swift/issues/292#issuecomment-554730055
- Have swiftlint in your Package.swift like we do on https://github.com/danger/swift/blob/master/Package.swift#L23
- Have a swiftlint binary in your repo and pass the path `SwiftLint.lint()` call
- Build swiftlint before running danger on the action

**Comment by NicholasTD07** [comment]
Running into this issue with a GitHub team account where in our repo, Issues is disabled since we don't use it...

**Comment by alok-dks** [comment]
Why would draft is not available on: pull_request. 
it should have the pull request object. 

```

/tmp/_tmp_dangerfile.swift:21:38: error: value of type 'GitHub.PullRequest' has no member 'draft'
7
    if let isPRDraft = danger.github.pullRequest.draft {
```

**Comment by 417-72KI** [comment]
@alok-dks 
Which version of Danger-Swift do you use on GitHub Actions?
 `draft` is available since [3.3.1](https://github.com/danger/swift/blob/master/CHANGELOG.md#331).

**Comment by alok-dks** [comment]
@417-72KI  I am using latest that GitHub Action that is our there from danger-swift people 
https://github.com/marketplace/actions/danger-swift
It is not `3.3.1`  as GitHub Action. 3.3.1 is latest danger-swift version. 

<img width="328" alt="Screen Shot 2020-12-08 at 10 58 01 PM" src="https://user-images.githubusercontent.com/70213416/101581989-d671c000-39a8-11eb-8c9d-3dd48c782128.png">
